CoreCLR tests build fails with `skipmanaged` option

Example of command used:
```
ROOTFS_DIR=`pwd`/.tools/rootfs/armel ./src/tests/build.sh -Release portablebuild=false -cross -armel -clang-10 -priority1 skipmanaged skipgeneratelayout skiprestorepackages /p:__SkipPackageRestore=true /p:UseSharedCompilation=false 
```

Errors:
```
/home/runtime/src/tests/Interop/DllImportSearchPaths/DllImportSearchPathsTest.csproj(21,5): error MSB3030: Could not copy the file "/home/runtime/artifacts/tests/coreclr/linux.armel.Release/Interop/DllImportSearchPaths/DllImportSearchPathsTest//DllImportSearchPathsTest.dll" because it was not found. [/home/runtime/src/tests/build.proj]
/home/runtime/src/tests/Interop/NativeLibrary/API/NativeLibraryTests.csproj(31,5): error MSB3030: Could not copy the file "/home/runtime/artifacts/tests/coreclr/linux.armel.Release/Interop/NativeLibrary/API/NativeLibraryTests//NativeLibraryTests.dll" because it was not found. [/home/runtime/src/tests/build.proj]
```

This is related to changes added to `DllImportSearchPathsTest.csproj` and `NativeLibraryTests.csproj` in #84663. With `skipmanaged` no managed dlls are built, that's why they can't be found.

cc @alpencolt

--- a/src/tests/Interop/DllImportSearchPaths/DllImportSearchPathsTest.csproj
+++ b/src/tests/Interop/DllImportSearchPaths/DllImportSearchPathsTest.csproj
@@ -8,17 +8,22 @@
     <CMakeProjectReference Include="../NativeLibrary/NativeLibraryToLoad/CMakeLists.txt" />
   </ItemGroup>
 
-  <Target Name="SetUpSubdirectory" AfterTargets="CopyNativeProjectBinaries">
-    <PropertyGroup>
-      <NativeLibrarySubdirectory>$(OutDir)/subdirectory</NativeLibrarySubdirectory>
-      <FileNameSuffix>-in-subdirectory</FileNameSuffix>
-    </PropertyGroup>
+  <PropertyGroup>
+    <LibrarySubdirectory>$(OutputPath)/subdirectory</LibrarySubdirectory>
+  </PropertyGroup>
+
+  <Target Name="SetUpSubdirectoryNative" AfterTargets="CopyNativeProjectBinaries">
     <ItemGroup>
-      <_FilesToCopy Include="$(OutDir)/$(TargetName).dll" />
       <_FilesToMove Include="$(OutDir)/libNativeLibrary.*" />
       <_FilesToMove Include="$(OutDir)/NativeLibrary.*" />
     </ItemGroup>
-    <Copy SourceFiles="@(_FilesToCopy)" DestinationFiles="@(_FilesToCopy -> '$(NativeLibrarySubdirectory)/%(Filename)%(Extension)')" />
-    <Move SourceFiles="@(_FilesToMove)" DestinationFiles="@(_FilesToMove -> '$(NativeLibrarySubdirectory)/%(Filename)%(Extension)')" />
+    <Move SourceFiles="@(_FilesToMove)" DestinationFiles="@(_FilesToMove -> '$(LibrarySubdirectory)/%(Filename)%(Extension)')"/>
+  </Target>
+
+  <Target Name="SetUpSubdirectoryManaged" AfterTargets="Build">
+    <ItemGroup>
+      <_FilesToCopy Include="$(OutDir)/$(TargetName).dll" />
+    </ItemGroup>
+    <Copy SourceFiles="@(_FilesToCopy)" DestinationFiles="@(_FilesToCopy -> '$(LibrarySubdirectory)/%(Filename)%(Extension)')"/>
   </Target>
 </Project>

--- a/src/tests/Interop/NativeLibrary/API/NativeLibraryTests.csproj
+++ b/src/tests/Interop/NativeLibrary/API/NativeLibraryTests.csproj
@@ -17,17 +17,23 @@
     <ProjectReference Include="$(TestSourceDir)Common/CoreCLRTestLibrary/CoreCLRTestLibrary.csproj" />
     <CMakeProjectReference Include="../NativeLibraryToLoad/CMakeLists.txt" />
   </ItemGroup>
+  <PropertyGroup>
+    <LibrarySubdirectory>$(OutputPath)/subdirectory</LibrarySubdirectory>
+    <FileNameSuffix>-in-subdirectory</FileNameSuffix>
+  </PropertyGroup>
 
-  <Target Name="SetUpSubdirectory" AfterTargets="CopyNativeProjectBinaries">
-    <PropertyGroup>
-      <NativeLibrarySubdirectory>$(OutDir)/subdirectory</NativeLibrarySubdirectory>
-      <FileNameSuffix>-in-subdirectory</FileNameSuffix>
-    </PropertyGroup>
+  <Target Name="SetUpSubdirectoryNative" AfterTargets="CopyNativeProjectBinaries">
     <ItemGroup>
       <AssembliesToCopy Include="$(OutDir)/libNativeLibrary.*" />
       <AssembliesToCopy Include="$(OutDir)/NativeLibrary.*" />
+    </ItemGroup>
+    <Copy SourceFiles="@(AssembliesToCopy)" DestinationFiles="@(AssembliesToCopy -> '$(LibrarySubdirectory)/%(Filename)$(FileNameSuffix)%(Extension)')" />
+  </Target>
+
+  <Target Name="SetUpSubdirectoryManaged" AfterTargets="Build">
+    <ItemGroup>
       <AssembliesToCopy Include="$(OutDir)/$(TargetName).dll" />
     </ItemGroup>
-    <Copy SourceFiles="@(AssembliesToCopy)" DestinationFiles="@(AssembliesToCopy -> '$(NativeLibrarySubdirectory)/%(Filename)$(FileNameSuffix)%(Extension)')" />
+    <Copy SourceFiles="@(AssembliesToCopy)" DestinationFiles="@(AssembliesToCopy -> '$(LibrarySubdirectory)/%(Filename)$(FileNameSuffix)%(Extension)')" />
   </Target>
 </Project>

